### PR TITLE
Update Development-Workflow document

### LINKED ISSUE #16991: Development Workflow is out of date (dashboard and workspace loader are now in separate repos)
### Is your task related to a problem? Please describe.

Since the Che dashboard and ws-loader are now separate images and separate repos, the Che 7 workflow doc needs updating.

https://github.com/eclipse/che/wiki/Development-Workflow,-Che-7

### Describe the solution you'd like

Updated doc! 

Since it's 7.14 (a hardening sprint) this is a perfect time to clean up doc that is now outdated with the updated repo/build structure.

### PR DESCRIPTION
**Pull request by skabashnyuk**
Fixes https://github.com/eclipse/che/issues/16991

### CODE REVIEW

File: Development-Workflow,-Che-7.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
-The best way to understand how a project works or to debug an issue is to get the source, built it, and run it locally. We document the best practices for developing improvements to the product. Eclipse Che has three major subsystems: the Che server, the agents running within a workspace, and the user dashboard (a JavaScript app). The workflow for each can be slightly different.
+The best way to understand how a project works or to debug an issue is to get the source, built it, and run it locally. We document the best practices for developing improvements to the product. Eclipse Che has several subsystems: the Che server, editors and plugins, and the user dashboard (a JavaScript app). The workflow for each can be slightly different.
```
Comment (metlos) [inline Development-Workflow,-Che-7.md]: ..., build it, ...

**Review by metlos — COMMENTED** [review]
Small typo found..

File: Development-Workflow,-Che-7.md [hunk 2]
```diff
@@ -18,9 +16,9 @@ To build the user dashboard submodule, you will need npm, bower, gulp, and pytho
   - yarn 1.16+
   - typings
 
-Learn more about how to [build the dashboard submodule here](https://github.com/eclipse/che/tree/master/dashboard).
+Learn more about how to [build the dashboard here](https://github.com/eclipse/che-dashboard/blob/master/README.md).
+Learn more about how to [build the Che Theia here](https://github.com/eclipse/che-theia/blob/master/README.md).
```
Comment (benoitf) [inline Development-Workflow,-Che-7.md]:
```suggestion
Learn more about how to [build the Che Theia here](https://github.com/eclipse/che-theia/blob/master/CONTRIBUTING.md).
```

?

File: Development-Workflow,-Che-7.md [hunk 3]
```diff
@@ -299,33 +219,22 @@ In order to keep the conversation clear and transparent, please limit discussion
...
 /che/samples                           # Code templates injected into new workspaces
 ```
 
 ### Repositories
 Some dependencies are managed in separate repositories as part of the `http://github.com/eclipse` organization. These dependencies are forks of other important projects, contain shared libraries that need to be managed on a different tagging lifecycle than Che, or have very large files within them (such as docs).
 ```
 https://github.com/eclipse/che-parent
-https://github.com/eclipse/che-lib
-https://github.com/eclipse/che-dev
+https://github.com/eclipse/che-dashaboard
```
Comment (benoitf) [inline Development-Workflow,-Che-7.md]:
```suggestion
https://github.com/eclipse/che-dashboard
```

**Pull request merged**